shell/platform/darwin/ios/framework/Source/platform_message_router.mm

```diff
@@ -20,10 +20,7 @@
   auto it = message_handlers_.find(message->channel());
   if (it != message_handlers_.end()) {
     FlutterBinaryMessageHandler handler = it->second;
-    NSData* data = nil;
-    if (message->hasData()) {
-      data = GetNSDataFromVector(message->data());
-    }
+    NSData* data = GetNSDataFromVector(message->data());
```

Comment:
This is a breaking change for anyone with a custom codec on iOS that has the same behavior. That may be nobody, but there's no way of knowing.

Reply:
Seems desirable for us to have the embedders behave consistently if we think this is likely to avoid surprising behaviour for users. i.e. both patch the codecs to handle empty data and patch the macOS embedder to send nil in that case. Reverted this bit.

Reply:
Looking at the [StandardMessageCodec docs](https://api.flutter.dev/flutter/services/StandardMessageCodec-class.html) I think we probably should patch the embedder as well, for the sake of consistency (and I should add docs for desktop).